src/legacy/core_plugins/kibana/server/lib/csp_usage_collector/csp_collector.ts

```diff
@@ -18,26 +18,25 @@
  */
 
 import { Server } from 'hapi';
-import { createCSPRuleString, DEFAULT_CSP_RULES } from '../../../../../server/csp';
+import { CspConfig } from '../../../../../../core/server';
 import { UsageCollectionSetup } from '../../../../../../plugins/usage_collection/server';
 
 export function createCspCollector(server: Server) {
   return {
     type: 'csp',
     isReady: () => true,
     async fetch() {
-      const config = server.config();
-
-      // It's important that we do not send the value of csp.rules here as it
-      // can be customized with values that can be identifiable to given
-      // installs, such as URLs
-      const defaultRulesString = createCSPRuleString([...DEFAULT_CSP_RULES]);
-      const actualRulesString = createCSPRuleString(config.get('csp.rules'));
+      const { strict, warnLegacyBrowsers, header } = server.newPlatform.setup.core.http.csp;
+      // This is used to get the default CSP header string.
+      const { header: defaultCspHeader } = new CspConfig();
```

Comment:
(related to my previous comment, I see why that was needed now): I think I still don't like the plugin having to instanciate CspConfig to get the default value. This is probably a NIT, but maybe having something like `http.defaultCsp` should be better in term of responsibilities? Also would allow to expose an interface instead of the concrete?